src/main/java/it/gov/pagopa/payhub/auth/service/user/IamUserInfoDTO2UserInfoMapper.java

```diff
@@ -115,4 +117,21 @@ private Broker getSessionBroker(IamUserInfoDTO iamUserInfoDTO, List<Operator> us
         return null;
     }
 
+    private void setBrokerInfo(UserInfo userInfo, String accessToken) {
+        IamUserInfoDTO iamUserInfo = tokenStoreService.load(accessToken);
+        if (iamUserInfo == null) {
+            throw new InvalidAccessTokenException("AccessToken not found");
+        }
+
+        List<Operator> userRoles = operatorsRepository.findAllByUserId(iamUserInfo.getInnerUserId());
+        Broker brokerInfo = getSessionBroker(iamUserInfo, userRoles, accessToken);
```

Comment:
all userRoles info should be also inside the userInfo.getOrganizations field, you don't need to retrieve again them